lib/utils/fields.js

```diff
@@ -35,3 +37,33 @@ export function fieldUnitBinTest(fields, col) {
     fields[col + 2].title == fields[col].title + ' bin'
   );
 }
+
+export function dataArrayToObject(
+  dataArray,
+  fields,
+  multivalueDelimiter = '; '
+) {
+  const dataObject = {};
+  dataArray.forEach((cell, idx) => {
+    if (cell === '' || cell == null) {
+      return;
+    }
+    const field = fields[idx];
+    let parsed;
+    if (field.multivalued) {
+      const split = cell.split(multivalueDelimiter);
+      parsed = split
+        .map((value) => parseDatatype(value, field.datatype))
+        .filter((parsed) => parsed !== undefined);
```

Comment:
So it is interesting that dataArrayToObject basically has built-in validation to some extent, by dropping invalid data that may be present in handsontable.

Reply:
Yeah, my general thought was that the output for this method should be able to to be serialized to JSON and validated against the original schema. I considered adding a `parseTypes` argument (default `true`) which would allow forgoing the parsing step and just give the raw string values from the table. But then I figured maybe it's better to keep it simple for now and add that in the future if it's needed.